### ISSUE TITLE
cilium hubble enable|disable are asymmetric

### ISSUE THREAD

**Issue by ensonic** [cilium/cilium#23777]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When I have cilium installed via helm with all defaults, I can use `cilium hubble enable --ui` to enable the hubble ui + relay. The confusing part is that I can't disable them using `cilium hubble disable --ui`.

As a workaround I've tried:
```
> cilium hubble enable --helm-set hubble.ui.enabled=false,hubble.relay.enabled=false
// Error: Unable to enable Hubble: services "hubble-peer" already exists
```

### Cilium Version

cilium-cli: v0.12.12 compiled with go1.19.4 on linux/amd64
cilium image (default): v1.12.5
cilium image (stable): v1.12.7
cilium image (running): v1.12.7

### Kernel Version

5.19.11-1rodete1-amd64 #1 SMP PREEMPT_DYNAMIC Debian 5.19.11-1rodete1 (2022-10-31) x86_64 GNU/Linux

### Kubernetes Version

Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.1", GitCommit:"8f94681cd294aa8cfd3407b8191f6c70214973a4", GitTreeState:"clean", BuildDate:"2023-01-18T15:58:16Z", GoVersion:"go1.19.5", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:43:11Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by nbusseneau** [comment]
Hello, thanks for the report! This is probably related to https://github.com/cilium/cilium-cli/issues/751. In a nutshell, the Cilium CLI does not interact with Helm to apply its changes, instead it does its own thing, so there are quirks around mixing both. We are aware of the issue and would very much love to fix it, but haven't been able to complete that work as of yet.

FWIW the command `cilium hubble disable` is supposed to disable both Hubble Relay and Hubble UI, no matter if `--ui` was specified or not at `cilium hubble enable` time. I haven't tried if that would work in your case (again, Helm quirks are possible), but it's worth a try!

**Comment by ensonic** [comment]
Yes `cilium hubble disable` work, but it also disabled hubble, right?

**Comment by nbusseneau** [comment]
> WHat is a recommended low-overhead default setup where the observability UI can be enabled/disabled as needed.

Either going full Helm or going full CLI (i.e. `cilium install` and then `cilium hubble ...` should work, unlike Helm install + `cilium hubble...`). AFAIK it's only mixing the two that causes issues, so you should be safe to stick with Helm rollouts to enable/disable Hubble without using `cilium hubble` commands.

> Maybe it is better to add a notes to the docs and call out that this does not yet work as you want?

I agree the docs are not that great in that regard, since the docs mostly recommend using `cilium install` and implicitly expect it will work, but of course there are still many users that use Helm instead (e.g. because of CI versioning). I think everybody kinda hoped the situation would be resolved fast enough for it not to be a huge issue, but it dragged on. If you see something specific that would benefit from improvements, may I suggest submitting a PR? It should be a quick win, that would be very appreciated :)

cc @tklauser in case there are complimentary comments.
